src/server.py

```diff
@@ -8,22 +8,20 @@
 def parse_request(request):
     """Parse and validate request to confirm parts are correct."""
     lines = request.split('\r\n')
-    try:
-        lines[1][:3] == 'GET'
-    except:
-        response_error(u'405 Method Not Allowed')
-        raise RuntimeError('Only accepts GET requests.')
-    try:
-        lines[1][-8:] == 'HTTP/1.1'
-    except:
-        response_error(u'505 HTTP Version Not Supported')
-        raise RuntimeError('Only accepts HTTP/1.1 protocol requests.')
-    try:
-        len(lines[1].split()) == 3 and lines[1][0] == '/'
-    except:
-        response_error(u'404 Page Not Found')
-        raise RuntimeError('URI not properly formatted.')
-    return lines[1]
+    words = lines[0].split()
+    if lines[0][:3] == 'GET':
```

Comment:
These `if/else` blocks can probably all be condensed into a single `if/elif/elif/.../else` block.